images/macos/scripts/build/install-codeql-bundle.sh

```diff
@@ -6,9 +6,24 @@
 
 source ~/utils/utils.sh
 
+# Retrieve the latest major version of the CodeQL Action to use in the base URL for downloading the bundle.
+releases=$(curl -s "https://api.github.com/repos/github/codeql-action/releases")
+
+# Get the release tags starting with v[0-9] and sort them in descending order, then parse the first one to get the major version.
+codeql_action_latest_major_version=$(echo "$releases" |
```

Comment:
Unconditionally updating to the next major version poses compatibility risks because major versions often bring breaking changes that need to be weighed in before putting them into the image. Can we leverage [toolset-XY.json](https://github.com/actions/runner-images/blob/79ad381addf50d2f8cc28e998e27d25894cac64f/images/macos/toolsets/toolset-15.json#L154) for specifying CodeQL version so we can have a control over it?

Reply:
Quick clarification: we only parse the latest major version of the Action in this line in order to populate the URL in the following lines, so that we can download the correct patch version of the CodeQL bundle. 

To address the larger concern, we currently run a series of CodeQL-specific canary test workflows during image rollouts (internal repo but I'm sure you know which one I'm talking about 😸). These workflows test the latest patch versions of the CodeQL Action & CodeQL CLI and serve to make sure that changes in CodeQL are not breaking the image; and also that other changes in the image are not breaking CodeQL.